File: GarageGo-BE/Models/SanPham.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GarageGo_BE.Models
{
    [Table("SANPHAM")]
    public class SanPham
    {
        [Key]
        [Column("MaSP")]
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        [Column("TenSP")]
        public string TenSanPham { get; set; } = string.Empty;

        [StringLength(500)]
        public string? MoTa { get; set; }

        [StringLength(255)]
        [Column("HinhAnh")]
        public string? HinhAnh { get; set; }

        [Required]
        [Column("DonGia", TypeName = "decimal(18,2)")]
        public decimal Gia { get; set; }

        [StringLength(20)]
        [Column("DonVi")]
        public string? DonVi { get; set; }

        [Column("SoLuongTon")]
        public int SoLuongTon { get; set; } = 0;
    }
}

File: GarageGo-BEJava/src/main/java/com/garagego/model/SanPham.java
package com.garagego.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "SANPHAM")
public class SanPham {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "MaSP")
    private Integer maSP;

    @Column(name = "TenSP", nullable = false, length = 100)
    private String tenSanPham;

    @Column(name = "MoTa", length = 500)
    private String moTa;

    @Column(name = "HinhAnh", length = 255)
    private String hinhAnh;

    @Column(name = "DonGia", nullable = false, precision = 18, scale = 2)
    private BigDecimal gia;

    @Column(name = "DonVi", length = 20)
    private String donVi;

    @Column(name = "SoLuongTon")
    private Integer soLuongTon = 0;
}
